pkg/apis/core/validation/validation_test.go

```diff
@@ -13049,6 +13081,7 @@ func TestValidateServiceUpdate(t *testing.T) {
 			tweakSvc: func(oldSvc, newSvc *core.Service) {
 				oldSvc.Spec.Type = core.ServiceTypeClusterIP
 				newSvc.Spec.Type = core.ServiceTypeLoadBalancer
+				newSvc.Spec.AllocateLoadBalancerNodePorts = utilpointer.BoolPtr(true)
```

Comment:
How about a case where we change this ?  I thought we only wanted to allow it to change when type becomes LB?  But I can't seem to find that

Reply:
Do you mean leaving `newSvc.Spec.AllocateLoadBalancerNodePorts` as empty here?

Reply:
Oh I see what you mean, added tests that changes `AllocateLoadBalancerNodePorts` from `true` to `false` and the other way around.